Replaces Resomi super "hearing"

PsiOmegaDelta opened the pull request:
The general idea of Resomi having sensitive hearing, able to pick up ques other species can only dream about is fine, but it needs to be handled in some other way.
Giving them the ability to see the names, equipment, and being able to examine mobs through solid matter is a step beyond mere hearing.

PsiOmegaDelta committed:
Removes Resomi super "hearing"

Resmi no longer have thermal vision, and can thus no longer see the names, equipment, or description of mobs through solid matter.

Aticius commented:
I'm generally against this.

It's current implementation is poor, yes. But, taking away something that's been there since their implementation  to take it away, without giving a proper replacement leaves a bad taste in my mouth.

Yes. I'm aware the current system is bad and needs improvement.

mustafakalash commented:
Would it be possible to use the icon override stuff to replace seeing people through walls with just seeing blips or similar?

PsiOmegaDelta committed:
Gives Resomi a Listen-In verb.

Gives them an approximate locations of all living mobs within their potential field of vision.

PsiOmegaDelta commented:
The Resomi lonesome hallucinations were taken away without anything else added.

As for icon overrides: Possible, tho said icons would still have the same name as the mob and we'd have to have every Resomi track every other mob to check if their within their field of vision or not, to apply the override under the proper conditions.

Resomi have been granted a verb that allow them to listen in to nearby movement. Would be doable to expand and have feedback to be affected by equipment to some degree.

martinlyra commented:
Wonderful! Can we remove wizard now? </s>

For a starter, consider using the `icons` in `client` (icons that are only visible to affected client) to add icons of _alive_ and _breathing_ that are _not_ in space and out of your view (use an `orange()`, or the square/block equivalent if any, to check if they are within the visible portions of the weldervision, regardless if they are in direct view or not).

Those said icons are not the mobs' icons, rather an amalgamate of the heart and lungs.

Edited for formatting.

MistakeNot4892 commented:
If there's any weight to the opinion of the original coder, this is better than thermal vision and personally I approve.

It may be worth considering removing or reducing the vision malus in view of this, though, since welder vision was chosen as a harsh balancing agent for permanent thermals.

MistakeNot4892 commented on code/modules/mob/living/carbon/human/species/station/resomi.dm:
```diff
@@ -85,8 +85,64 @@
-	else
-		return SEE_SELF
+/datum/species/resomi/handle_post_spawn(var/mob/living/carbon/human/H)
+	H.verbs |= /mob/living/carbon/human/proc/resomi_hearing
```
There's a species var called inherent_verbs that handles adding and removing verbs.

PsiOmegaDelta commented:
I have no objection against any welder hud adjustments one way or another.
Could keep it, remove it, or borrow tg's full-screen overlay that's basically the total-blackness area but without the dotted semi-see through parts.

Raptor1628 commented:
Whatever balancing that needs to be done to work around thermals being removed can be done, but removing thermals has been a long time coming. The replacement system seems reasonable.

LorenLuke commented:
Couple things:
- think about how actual hearing works. If something's on the other side of the wall, be it speech or anything with a auditory 'visible message' (you hear metal clanking). Allow them to hear everything within range.
- With hallucinations, you can mouse over pun pun and see the name, maybe have blips be secondary and client side (or however they work), and have the blinking blips labeled as just be that... Person, silicon, etc. However, just have them be a solid color, and when someone speaks always have them be able to hear them. (i.e. if you have ANYTHING making a noise, it's showing up as a blinking blip. Have all spoken words be labeled with a name.)
-  given the above, have solid blips be present only when listening, that is, a sound is not necessary to get a blip when you're using the link verb. These blips should also overlay to the effects of the any stealth gear, as well as in darkness.
- extend Resomii whisper hearing range to three tiles, and a full screen when using a verb.

Just a couple thoughts and suggestions.

Edit: Additionally, not seeing giant stompy mechs with Resomi vision is silly.

MistakeNot4892 commented:
Alternatively, just give them a directional indicator whenever the mob is passed a sound via playsound(). That would give them stompy mech footstep indicators, it'd let them hear fights and gunshots, and it would stop them knowing exactly what is causing the noise.

mustafakalash commented:
Isn't there a stub of an action/noise->text system for blind folk?

PsiOmegaDelta commented:
Most of the suggestions involving say/show_message tweaks are likely to be server-performance-killers before #12123 is done.

Client images cannot be seen unless what said images are attached to can also be seen. Thus we need to use either alternative-image-appearances conditionally or to pixel shift an image attached to the Resomi into place, and both will require that each Resomi listen to all movements of every other mob.

BlueNexus commented:
Status on this? Could use it merged at some point.

PsiOmegaDelta commented:
Still need to do something about the vision overlay, and add a system to allow more detailed sonar feedback (based on equipment, etc.).

Aticius commented:
If it hasn't been found out- This has my good graces now (if they even matter!). Removing thermals would be super nice- replacing it with a mild Oxyloss overlay (or something similar) would be better.

PsiOmegaDelta commented:
Per IRC discussion Aticius would like to have these changes in as soon as possible, with other feature tweaks coming later, due to increasing controversy over thermal vision.
